tripal/src/Hook/TripalViewsHooks.php

```diff
@@ -70,4 +34,45 @@ public function viewPresave(ViewEntityInterface $view): void {
     }
   }
 
+  /**
+   * Implements hook_views_query_alter().
+   *
+   * Alters the Tripal Content Type Listing view to filter content types
+   * based on the current user's permissions.
+   */
+  #[Hook('views_query_alter')]
+  public function viewsQueryAlter(ViewExecutable $view, QueryPluginBase $query) {
+    // Only alter the Tripal Content Type Listing view.
+    if ($view->id() !== 'tripal_content_type_listing') {
+      return;
+    }
+
+    // Load all Tripal entity types.
+    $entity_types = \Drupal::service('entity_type.manager')
+      ->getStorage('tripal_entity_type')
+      ->loadByProperties([]);
```

Comment:
It's not performant to load all the entity types for this. It's better to use `TripalEntityAccessControlHandler::getTripalContentPermissionsList('view');` to get the list of permissions. This also allows us to not make any assumptions about the format of the permission string.